### PR TITLE
[DOC] Document setting environment variable

### LINKED ISSUE #759: Document support for `variables` in environment spec
### Context

In https://github.com/conda-incubator/conda-store-ui/pull/354, we add support for passing environment variables in the `conda-store-ui` through the YAML editor.
This allows users to create envs for GPU-powerd workflows, by adding the `CONDA_OVERRIDE_CUDA`.
This needs to be documented in https://conda.store/conda-store-ui/tutorials/create-envs#graphical-interface, with a new 3rd level heading after the pip section.

### Value and/or benefit

Documenting available features for conda-store users.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pavithraes**
Fixes #759

## Description

This pull request:

- Adds a note about setting environment variable
- Minor misc clean-up

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

## Additional information

Merge **after** the conda-store-ui feature is merged.

### CODE REVIEW

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:89]:
```suggestion
This feature is available *after* conda-store-ui version 2024.1.1.
```

File: docusaurus-docs/conda-store-ui/tutorials/setup.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/setup.md:47]: Please move changes in this file into a separate PR. To me, these look unrelated to the environment variables change, which means they can be merged separately (and likely much faster as well).

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:98]:
```suggestion
    CONDA_OVERRIDE_CUDA: '12.0'
```

I suggest making this change because it makes it clear that the value is a string.

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:91]: I'd link to the official conda documentation here as well, see:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#overriding-detected-packages

Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:91]:
```suggestion
Currently, only the `CONDA_OVERRIDE_CUDA` environment variable can be specified, which allows setting the CUDA version for building packages with GPU support.
```
Proposed alternative wording, which is a bit more specific.

**Review by nkaretnikov — CHANGES_REQUESTED** [review]
@pavithraes Thanks for documenting this! I've left some comments for your consideration.

Note: this feature is already in conda-store-server, but it's still in review in conda-store-ui. So I'm going to mark this PR as blocked until that one lands, see: https://github.com/conda-incubator/conda-store-ui/pull/354.

File: docusaurus-docs/conda-store-ui/tutorials/setup.md
Comment (pavithraes) [reply docusaurus-docs/conda-store-ui/tutorials/setup.md:47]: Thanks for your review :)

I did mention in the PR description that it includes some minor, miscellaneous clean-up, so I thought this would be alright. 

However, I do agree we should aim for cleaner+specific PRs. I'll revert these updates.

**Comment by pavithraes** [comment]
@nkaretnikov Thanks for your review, I have addressed all your comments. :)

Please take another look and approve it when you get a chance. Of course, even after approval, we'll only merge after `https://github.com/conda-incubator/conda-store-ui/pull/354` is merged.

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:95]:
```suggestion
You can set environment variables in the YAML editor with the following syntax:
```

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [inline docusaurus-docs/conda-store-ui/tutorials/create-envs.md:99]:
```suggestion
  CONDA_OVERRIDE_CUDA: '12.0'
```

Comment (nkaretnikov) [reply docusaurus-docs/conda-store-ui/tutorials/create-envs.md:99]: This is just because our YAML pretty-printer uses 2 spaces by default.

File: docusaurus-docs/conda-store-ui/tutorials/create-envs.md
Comment (nkaretnikov) [reply docusaurus-docs/conda-store-ui/tutorials/create-envs.md:95]: This is just to have consistent spelling, it's "YAML editor" in other parts of the doc.

**Review by nkaretnikov — APPROVED** [review]
@pavithraes Thanks, LGTM! I've left two comments, which would be nice to fix, but those are not blockers. OK to merge this after the companion UI PR lands: https://github.com/conda-incubator/conda-store-ui/pull/354

**Pull request merged**
